R/Resampling.R

```diff
@@ -168,19 +180,24 @@ Resampling = R6Class("Resampling",
       task = assert_task(as_task(task))
       strata = task$strata
       groups = task$groups
+      weights = task$weights_resampling$weight
+
+      use_weights = "weights" %in% self$properties && !isFALSE(self$param_set$values$use_weights) && !is.null(weights)
+
+      if (sum(!is.null(strata), !is.null(groups), use_weights) >= 2L) {
+        stopf("Stratification, grouping and weighted resampling are mutually exclusive")
```

Comment:
Why do we have this restriction? At least for groups we could sum up the weights and also for strata we could also use the weights within each stratum?